src/Sushi.Mediakiwi.API/Controllers/AssetController.cs

```diff
@@ -0,0 +1,271 @@
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Returns all galleries that are available for the supplied user
+        /// </summary>
+        /// <returns></returns>
+        [Produces("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [HttpGet("GetGalleries")]
+        public async Task<ActionResult<List<ListItemCollectionOption>>> GetGalleries()
+        {
+            var temp = new List<ListItemCollectionOption>();
+
+            var galleries = await Data.Gallery.SelectAllAccessibleAsync(MediakiwiUser);
+            foreach (var gallery in galleries)
+            {
+                temp.Add(new ListItemCollectionOption() 
+                { 
+                    Text = gallery.CompletePath, 
+                    Value = gallery.ID.ToString(), 
+                    IsEnabled = true 
+                });
+            }
+
+            return Ok(temp);
+        }
+
+        /// <summary>
+        /// Saves the supplied asset to Azure and returns information about the created/updated asset
+        /// </summary>
+        /// <param name="request">Contains all information about the asset</param>
+        /// <returns></returns>
+        [Produces("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [HttpPost("SaveAsset")]
+        public async Task<ActionResult<SaveAssetResponse>> SaveAsset([FromForm] SaveAssetRequest request)
+        {
+            // Request is empty
+            if (request == null)
+            {
+                return BadRequest();
+            }
+
+            // New image
+            if (request.ID.GetValueOrDefault(0) == 0)
+            {
+                // But no data supplied
+                if (request.Data?.Length == 0 || string.IsNullOrWhiteSpace(request.Data?.FileName) == true)
+                {
+                    return BadRequest();
+                }
+
+                // But no gallery supplied
+                if (request.GalleryID.GetValueOrDefault(0) == 0)
+                {
+                    return BadRequest();
+                }
+            }
+
+            // The target asset
+            Data.Asset targetAsset = new Data.Asset();
+
+            // We're updating an asset, get it from DB
+            if (request.ID.GetValueOrDefault(0) > 0)
+            {
+                targetAsset = await Data.Asset.SelectOneAsync(request.ID.Value);
+            }
+
+            // We have new data provided, process it
+            if (request.Data != null)
+            {
+                targetAsset.Type = request.Data.ContentType;
+                targetAsset.Extention = request.Data.FileName.Substring(request.Data.FileName.LastIndexOf('.'));
```

Comment:
LastIndexOf might return -1 if the filename does not contain a dot (.)
Would this not lead to an exception if fed to the Substring function?

Reply:
it would, but posting a file without extension is not allowed, so will throw a BadRequest when that happens